sdks/v4-sdk/src/PositionManager.ts

```diff
@@ -276,25 +284,34 @@ export abstract class V4PositionManager {
 
     // If migrating, we need to settle and sweep both currencies individually
     if (isMint(options) && options.migrate) {
-      // payer is v4 positiion manager
-      planner.addSettle(position.pool.currency0, false)
-      planner.addSettle(position.pool.currency1, false)
-      planner.addSweep(position.pool.currency0, options.recipient)
-      planner.addSweep(position.pool.currency1, options.recipient)
+      if (options.useNative) {
+        // unwrap the exact amount needed to send to the pool manager
+        planner.addUnwrap(OPEN_DELTA)
+        // payer is v4 position manager
+        planner.addSettle(position.pool.currency0, false)
+        planner.addSettle(position.pool.currency1, false)
+        // sweep any leftover wrapped native that was not unwrapped
+        planner.addSweep(position.pool.currency0.wrapped, options.recipient)
+        planner.addSweep(position.pool.currency1, options.recipient)
+      } else {
+        // payer is v4 position manager
+        planner.addSettle(position.pool.currency0, false)
+        planner.addSettle(position.pool.currency1, false)
+        planner.addSweep(position.pool.currency0, options.recipient)
+        planner.addSweep(position.pool.currency1, options.recipient)
+      }
     } else {
       // need to settle both currencies when minting / adding liquidity (user is the payer)
       planner.addSettlePair(position.pool.currency0, position.pool.currency1)
```

Comment:
think you can just add this if case here and delete it from below


```suggestion
      planner.addSettlePair(position.pool.currency0, position.pool.currency1)
      // Any sweeping must happen after the settling.
      let value: string = toHex(0)
     // When not migrating and adding native currency, add a final sweep
     if (options.useNative && !(isMint(options) && options.migrate)) {
     value = toHex(amount0Max)
     // native currency will always be currency0 in v4
     planner.addSweep(position.pool.currency0, MSG_SENDER)
    }
```

Reply:
nice!